python/paddle/fluid/initializer.py

```diff
@@ -165,7 +166,8 @@ def __call__(self, var, block):
                 'force_cpu': self._force_cpu or force_init_on_cpu()
             },
             stop_gradient=True)
-        var.op = op
+        if not base.enabled():
+            var.op = op
```

Comment:
why imperative mode doesn't allow it?

Reply:
because var will refer to op and op will not be removed forever.

Reply:
why? I thought var and op will be deleted together after backward

Reply:
Weight will keep in block and not delete, so the initializer will not be removed too.